deephaven/web-client-ui
# feat: Add support for multi-partition parquet:kv tables

Added feature to `IrisGridPartitionSelector` to display additional partition selector dropdowns when more than one partition column is present. The values displayed in the dropdown depend on the selected partitions of columns to the left of it.

The "Append Command" button and associated functions are removed in preparation for #1143. 

Closes #1438

#### Testing instructions:
1. Create and open a multi-partition table:
```python
from deephaven import empty_table

part = empty_table(4).update("II=ii")

from deephaven.parquet import write, read

write(part, "/tmp/pt-1438/x=0/y=0/part.parquet")
write(part, "/tmp/pt-1438/x=0/y=1/part.parquet")
write(part, "/tmp/pt-1438/x=1/y=0/part.parquet")
write(part, "/tmp/pt-1438/x=1/y=1/part.parquet")

partition_table = read("/tmp/pt-1438")
```
2. Use the dropdown to select any combination of partitions and check that it updates the table correctly.
    * The dropdown should only display values that are available after filtering for all the partitions left  of it. (E.g. A state/province dropdown should only contain the states/provinces corresponding to the selected country.)
3. Test filtering by typing directly in the dropdown and searching by using the search bar above the list of available partitions.
4. Retest using other data types for partition columns (dates, strings, chars, doubles). E.g:
```python
from deephaven.parquet import write, read

write(part, "/tmp/pt2-1438/date_partition=2023-08-01/id_partition=id-001/decimal=0.1/part.parquet")
write(part, "/tmp/pt2-1438/date_partition=2023-08-01/id_partition=id-002/decimal=0.15/part.parquet")
write(part, "/tmp/pt2-1438/date_partition=2023-08-02/id_partition=id-001/decimal=0.08/part.parquet")
write(part, "/tmp/pt2-1438/date_partition=2023-08-02/id_partition=id-002/decimal=0.08/part.parquet")
write(part, "/tmp/pt2-1438/date_partition=2023-08-02/id_partition=id-003/decimal=0.1/part.parquet")

partition_table = read("/tmp/pt2-1438")
```
